### PR TITLE
Add end to end test

### PR DESCRIPTION
**Pull request by rowanhemsi**
Add end to end test to package, with [related utils from xlsxwriter](https://github.com/jmcnamara/XlsxWriter/blob/main/xlsxwriter/test/excel_comparison_test.py). Also fix issues that came to light with this test:
* inconsistent parameter name in api functions (`contentsheet`/`contentsheet_label`)
* integration issue between `auto_width` and links/rich text for Python 3.6 and 3.7

Also increase breadth of CI testing following CI issues

### CODE REVIEW

File: gptables/core/api.py [hunk 1]
```diff
@@ -101,7 +101,8 @@ def write_workbook(
         sheets,
         theme = None,
         cover = None,
-        contentsheet = "Contents",
+        contentsheet = None,
```
Comment (alexander-newton) [inline gptables/core/api.py:104]: document contentsheet here - note that it is deprecated and shouldn't exist

File: gptables/core/wrappers.py [hunk 2]
```diff
@@ -944,19 +945,22 @@ def _longest_line_length(cell_val):
 |\r\n|\n"""
 
         if isinstance(cell_val, str):
-            return(max([len(line) for line in re.split(split_strings, cell_val)]))
+            max_length = max([len(line) for line in re.split(split_strings, cell_val)])
```
Comment (alexander-newton) [inline gptables/core/wrappers.py:948]: eventually would be nice to use case when from 3.10 - not needed now

File: gptables/test/test_utils/excel_comparison_test.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
+###############################################################################
```
Comment (alexander-newton) [inline gptables/test/test_utils/excel_comparison_test.py:1]: can we add a neater join of this onto the backlog - figure out how to make sure we're up to date

**Review by alexander-newton — APPROVED** [review]
happy with this

File: gptables/test/test_utils/excel_comparison_test.py (hunk 3 above)
Comment (rowanhemsi) [reply gptables/test/test_utils/excel_comparison_test.py:1]: Ticket in the backlog to improve how we access these files

File: gptables/core/api.py (hunk 1 above)
Comment (rowanhemsi) [reply gptables/core/api.py:104]: Documented with note and in `Parameters`

**Pull request merged**